src/viam/ur/module/utils.hpp

```diff
@@ -10,19 +12,18 @@ void configure_logger(const viam::sdk::ResourceConfig& cfg);
 
 // helper function to extract an attribute value from its key within a ResourceConfig
 template <class T>
-T find_config_attribute(const viam::sdk::ResourceConfig& cfg, const std::string& attribute) {
+std::optional<T> find_config_attribute(const viam::sdk::ResourceConfig& cfg, const std::string& attribute) {
```

Comment:
Regarding the case for the lack of `value_or` that doesn't eagerly evaluate, you could also have:

```
template<class T, class Callable>
T find_config_attribute(const Resourceconfig& config, const std::string& attribute, Callable&& c) {
    auto result = find_config_attribute(config, attribute);
    if (result) return *result;
    return std::forward<C>(c)();
}
```
Where you can pass in some lambda to compute the default. Then:

```
current_state_->output_csv_dir_path = find_config_attribute<std::string>(cfg, "csv_output_path", [] {
    // deal with VIAM_MODULE_DATA stuff
});
```
But I'll understand if you think that's taking it too far.